### PR TITLE
Update CSL to protocol revision 0x8adc5fa2448cb65e

### PR DESCRIPTION
**Pull request by reet-**
- Move CSL module to bsbsc directory
- Make CSL file ops overridable
- Get rid of `CMD_CHECK_VERSION`
- Get rid of `CMD_META_DATA`
- Skip vendor commands
- Support one-byte `CMD_WRITE`s
- Adjust check string null termination handling

### CODE REVIEW

File: grub-core/loader/bsbsc/csl.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/* csl.h - Command Stream Loader (CSL) header */
+/*
+ *  GRUB  --  GRand Unified Bootloader
+ *  Copyright (C) 2019  codelabs GmbH
```
Comment (Kensan) [inline grub-core/loader/bsbsc/csl.h]: 2020?

File: grub-core/loader/bsbsc/csl.c [hunk 2]
```diff
@@ -376,8 +346,13 @@ csl_dispatch (const grub_file_t file,
 			return csl_cmd_set_entry_point (file, (grub_size_t) length);
 		case CMD_CHECK_CPUID:
 			return csl_cmd_check_cpuid (file, (grub_size_t) length);
-		case CMD_META_DATA:
-			/* We do not care about stream meta data */
+		case VENDOR_CMD_ID_START ... VENDOR_CMD_ID_END:
+			/* avoid fseek. otherwise SBS module must implement it. */
```
Comment (Kensan) [inline grub-core/loader/bsbsc/csl.c:350]: Capitalization.

File: grub-core/loader/bsbsc/csl.c [hunk 3]
```diff
@@ -436,11 +411,18 @@ csl_cmd (grub_extcmd_context_t ctxt __attribute__ ((unused)),
 	file = csl_fs_ops.open (argv[0], GRUB_FILE_TYPE_NONE);
 	if (! file)
 		return grub_errno;
-	if (csl_fs_ops.size (file) % 8 != 0)
+
+	if (csl_fs_ops.read (file, &magic, 8) != 8)
```
Comment (Kensan) [inline grub-core/loader/bsbsc/csl.c:415]: Just to double-check: the underlying read implementation already handles the case where the file is less than 8 bytes in size? So magic is always initialized to a sane value even when a read error occurs?

**Review by Kensan — CHANGES_REQUESTED** [review]
Just some minor comments.

File: grub-core/loader/bsbsc/csl.c (hunk 2 above)
Comment (reet-) [reply grub-core/loader/bsbsc/csl.c:350]: I use lowercase comments in C, see also line 283.

File: grub-core/loader/bsbsc/csl.c (hunk 3 above)
Comment (reet-) [reply grub-core/loader/bsbsc/csl.c:415]: When a read error occurs here, an error is returned and processing stops. The `magic` variable is not accessed. The underlying `read` impl. is either `grub_file_read` or the upcoming `sbs_read`, both return < 8 if less bytes are available to read.

**Review by Kensan — APPROVED** [review]
Lgtm

**Pull request merged**